proposals/0026-hlsl-long-vector-type.md

```diff
@@ -0,0 +1,278 @@
+<!-- {% raw %} -->
+
+# HLSL Long Vectors
+
+* Proposal: [0026-HLSL-Vectors](0026-hlsl-vector-type.md)
+* Author(s): [Anupama Chandrasekhar](https://github.com/anupamachandra), [Greg Roth](https://github.com/pow2clk)
+* Sponsor: [Greg Roth](https://github.com/pow2clk)
+* Status: **Under Consideration**
+
+## Introduction
+
+HLSL has always supported vectors of as many as four elements of different element types (int3, float4, etc.).
+These are useful in a traditional graphics context for representation and manipulation of
+ geometry and color information.
+The evolution of HLSL as a more general purpose language targeting Graphics and Compute
+ greatly benefit from longer vectors to fully represent these operations rather than to try to
+ break them down into smaller constituent vectors.
+This feature adds the ability to declare and use native HLSL vectors longer than four elements.
```

Comment:
I don't understand the distinction between declaration and use.  I think this line should just be simplified to add native support like so:
```suggestion
This feature introduces native support for vectors exceeding four elements.
```

Reply:
Declaration without use in intrinsics isn't terribly useful of course, but there is a distinction between them. I drew the distinction because both declaration and use are limited in certain ways. The rest of the spec addresses them individually and this set that up. 

I don't feel too strongly about it for the introduction though.

Reply:
It would be really odd to have the language allow you to declare something that you then can't use. I might instead phrase this something like:

> This feature allows declaring objects of vector type with more than four elements, and defines how those objects may be used.